k-distribution/include/kframework/builtin/domains.md

```diff
@@ -533,10 +1082,43 @@ module INT
   syntax Int ::= randInt(Int) [function, hook(INT.rand)]
   syntax K ::= srandInt(Int) [function, hook(INT.srand)]
 endmodule
-
+```
+
+IEEE 754 Floating-point Numbers
+-------------------------------
+
+Provided here is the syntax of an implementation of arbitrary-precision
+floating-point arithmetic in K based on a generalization of the IEEE 754
+standard. This type is hooked to an implementation of floats provided by the 
+backend.
+
+The syntax of ordinary floating-point values in K consists of an optional sign
+(+ or -) followed by an optional integer part, followed by a decimal point,
+followed by an optional fractional part. Either the integer part or the 
+fractional part must be specified. The mantissa is followed by an optional
+exponent part, which consists of an `e` or `E`, an optional sign (+ or -), 
+and an integer. The expoennt is followed by an optional suffix, which can be
+either `f`, `F`, `d`, `D`, or `pNxM` where `N` and `M` are positive integers.
+`p` and `x` can be either upper or lowercase.
+
+The value of a floating-point literal is computed as follows: First the
+mantissa is read as a rational number. Then it is multiplied by 10 to the
+power of the exponent, which is interpreted as an integer, and defaults to
+zero if it is not present. Finally, it is rounded to the nearest possible
+value in a floating-point type represented like an IEEE754 floating-point type,
+with the number of bits of precision and exponent specified by the suffix.
+A suffix of `f` or `f` represents the IEEE `binary32` format. A suffix of `d`
+or `D`, or no suffix, represents the IEEE `binary64` format. A suffix of
+`pNxM` (either upper or lowercase) specifies exactly `N` bits of precision and
+`M` bits of exponent. The number of bits of precision is assumed to include
+any optional `1` that precedes the IEEE 754 mantissa. In other words, `p24x8`
+is equal to the IEEE `binary32` format, and `p53x11` is equal to the IEEE
+`binary64` format.
+
+```k
 module FLOAT-SYNTAX
   syntax Float [hook(FLOAT.Float)]
-  syntax Float ::= r"([\\+-]?[0-9]+(\\.[0-9]*)?|\\.[0-9]+)([eE][\\+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]d+))?([fFdD]|([pP][0-9]+[xX][0-9]+))?" [token, prec(1)]
+  syntax Float ::= r"([\\+-]?[0-9]+(\\.[0-9]*)?|\\.[0-9]+)([eE][\\+-]?[0-9]+)?([fFdD]|([pP][0-9]+[xX][0-9]+))?" [token, prec(1)]
```

Comment:
What's going on here? Are we removing support for some class of floating point parses?

Reply:
Removed here is the case when the exponent has a decimal point, which I'm quite certain was never actually working and should never have been part of this syntax.